drivers/ksz8851snl/ksz8851_netdev.c

```diff
@@ -0,0 +1,622 @@
+/*
+ * Copyright (C) 2015 Ell-i open source co-operative
+ *                    Kaspar Schleiser <kaspar@schleiser.de>
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup     driver_ksz8851snl
+ * @{
+ *
+ * @file
+ * @brief       Internal functions for the ksz8851snl driver
+ *
+ * @author      Kaspar Schleiser <kaspar@schleiser.de>
+ *
+ * @}
+ */
+
+#include <assert.h>
+#include <errno.h>
+
+#include "mutex.h"
+#include "ksz8851snl_regs.h"
+#include "ksz8851snl.h"
+
+#include "xtimer.h"
+
+#include "net/netdev2.h"
+#include "net/eui64.h"
+#include "net/ethernet.h"
+
+#define ENABLE_DEBUG (1)
+#include "debug.h"
+
+#define SECT_NAME "[KSZ8851 Driver]:"
+
+static void lock(ksz8851snl_t *dev) {
```

Comment:
also makes sense to inline this function, but might be that the compililer does this anyway...